cmake/FindLibxc.cmake

```diff
@@ -0,0 +1,46 @@
+###############################################################################
+# - Find ELPA
+# Find the native ELPA headers and libraries.
+#
+#  ELPA_FOUND        - True if libelpa is found.
+#  ELPA_LIBRARIES    - List of libraries when using libyaml
+#  ELPA_INCLUDE_DIR - Where to find ELPA headers.
+#
+
+#find_path(ELPA_INCLUDE_DIR
+#    elpa/elpa.h
+#    HINTS ${ELPA_DIR}
+#    PATH_SUFFIXES "include" "include/elpa"
+#    )
+
+include(FindPackageHandleStandardArgs)
+
+find_package(PkgConfig)
+
+if(PKG_CONFIG_FOUND)
+  pkg_search_module(Libxc REQUIRED IMPORTED_TARGET GLOBAL libxc)
+  find_package_handle_standard_args(Libxc DEFAULT_MSG Libxc_LINK_LIBRARIES Libxc_INCLUDE_DIRS)
+else()
+  find_package(Libxc REQUIRED HINTS
+    ${Libxc_DIR}/share/cmake/Libxc
+    ${Libxc_DIR}/lib/cmake/Libxc
+    ${Libxc_DIR}/lib64/cmake/Libxc
```

Comment:
Shall the default CMake interface of libxc being used, and pkg-config as a fallback method?

Reply:
OK, though a little complicated logic needs to be used here. Lines like these would still work:
```
# NO REQUIRED here, otherwhile it would throw error
# with no LibXC found.
find_package(Libxc HINTS
    ${Libxc_DIR}/share/cmake/Libxc
    ${Libxc_DIR}/lib/cmake/Libxc
    ${Libxc_DIR}/lib64/cmake/Libxc
  )
if(NOT Libxc_FOUND AND PKG_CONFIG_FOUND)
  pkg_search_module(Libxc REQUIRED IMPORTED_TARGET GLOBAL libxc)
  find_package_handle_standard_args(Libxc DEFAULT_MSG Libxc_LINK_LIBRARIES Libxc_INCLUDE_DIRS)
endif()

# Copy the results to the output variables and target.
# if find_package() above works, Libxc::xc would be present and
# below would be skipped.
if(Libxc_FOUND AND NOT TARGET Libxc::xc)
        set(Libxc_LIBRARY ${Libxc_LINK_LIBRARIES})
        set(Libxc_LIBRARIES ${Libxc_LIBRARY})
        set(Libxc_INCLUDE_DIR ${Libxc_INCLUDE_DIRS})
        add_library(Libxc::xc UNKNOWN IMPORTED)
        set_target_properties(Libxc::xc PROPERTIES
                IMPORTED_LOCATION "${Libxc_LIBRARY}"
                INTERFACE_INCLUDE_DIRECTORIES "${Libxc_INCLUDE_DIR}")
else if(NOT Libxc_FOUND)#No pkg_config
  message(FATAL_ERROR
          "LibXC : We need pkg-config to get all information about the libxc library")
endif()
```